src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -229,6 +230,11 @@ internal CookieContainer CookieContainer
             set
             {
                 CheckDisposedOrStarted();
+                if (value == null)
+                {
+                    throw new ArgumentNullException("CookieContainer", SR.net_http_invalid_cookiecontainer);
```

Comment:
The reason I included this null check  is that msdn says [HttpClientHandler  *will throw* an exception when cookie container is null.](https://msdn.microsoft.com/en-us/library/system.net.http.httpclienthandler.cookiecontainer(v=vs.110).aspx). If it is fine that it throws an exception in the sendAsync, I will remove it from here